contracts/staking/compound/GoodCompoundStaking.sol

```diff
@@ -85,31 +82,59 @@ contract GoodCompoundStaking is SimpleStaking {
 		override
 		returns (address, uint256)
 	{
+		ERC20 comp = ERC20(nameService.getAddress("COMP"));
+		uint256 compBalance = comp.balanceOf(address(this));
+		address daiAddress = nameService.getAddress("DAI");
+		Uniswap uniswapContract =
+			Uniswap(nameService.getAddress("UNISWAP_ROUTER"));
+		uint256 daiFromComp;
+		cERC20 cToken = cERC20(address(iToken));
+		address[] memory path = new address[](2);
+		if (compBalance > 0) {
+			path[0] = address(comp);
+			path[1] = daiAddress;
+			comp.approve(address(uniswapContract), compBalance);
+			uint256[] memory compSwap =
+				uniswapContract.swapExactTokensForTokens(
+					compBalance,
+					0,
+					path,
+					address(this),
+					block.timestamp
+				);
+			daiFromComp = compSwap[1];
+		}
```

Comment:
i'm thinking we should extract uniswap code to a helper library this will also help reduce reserve size